app/models/manageiq/providers/azure/cloud_manager/refresh_parser.rb

```diff
@@ -235,27 +249,29 @@ def add_instance_disk(disks, size, name, location)
         super(disks, size, name, location, "azure")
       end
 
-      def populate_hardware_hash_with_networks(hardware_networks_array, instance)
-        nics = instance.fetch_path('properties', 'networkProfile', 'networkInterfaces')
+      def populate_hardware_hash_with_networks(networks_array, instance)
```

Comment:
Rewrote the whole implementation. The armrest gem no longer queries network ip addresses and inserts to the vm object. Make explicit calls here to get them.